### PR TITLE
Created hash function in UserRepository (Closes #29)

### LINKED ISSUE #29: Hashing passwords
Create method that will take raw password and seed, to return hashed one.

### PR DESCRIPTION
**Pull request by dvasyab**
Created function to hash passwords to provide more security to database. It takes raw password and email as seed. Some of tests for UserRepository was changed.

### CODE REVIEW

File: src/main/java/shortener/httphandler/UserController.java [hunk 1]
```diff
@@ -53,7 +53,7 @@
     }
 
     try {
-      userRepository.create(userEmail, userPassword);
+      userRepository.create(userEmail, userRepository.hashFunc(userPassword, userEmail));
```
Comment (IINamelessII) [inline src/main/java/shortener/httphandler/UserController.java:56]: call the function outside of the "try" block if you are not expecting exceptions in it

File: src/main/java/shortener/users/UserRepository.java [hunk 2]
```diff
@@ -14,6 +16,39 @@
...
+   * @param rawPassword not hashed password
+   * @param emailAddress email address used as seed
+   * @return hashed password
+   */
+  public String hashFunc(String rawPassword, String emailAddress) {
+    try {
+      MessageDigest sha256 = MessageDigest.getInstance("SHA-256");
+
+      final String salt = emailAddress;
+      final String localSeed = "AmCDmG";
```
Comment (IINamelessII) [inline src/main/java/shortener/users/UserRepository.java:31]: Extract this seed into class'es const and use it here
Check validation of users creation for example (regexp patterns)

**Review by IINamelessII — CHANGES_REQUESTED** [review]

File: src/main/java/shortener/users/UserRepository.java (hunk 2 above)
Comment (RedBullDogE) [reply src/main/java/shortener/users/UserRepository.java:31]: Probably we'll move this function into separate class such as utils

File: src/main/java/shortener/httphandler/UserController.java (hunk 1 above)
Comment (dvasyab) [reply src/main/java/shortener/httphandler/UserController.java]: took out hashFunc() before the try block

**Review by IINamelessII — APPROVED** [review]

**Comment by IINamelessII** [comment]
@dvasyab never user "last" as commit message :) You see why)

**Comment by yugisu** [comment]
@dvasyab Could you have a look at the failed build so we would be able to proceed to merging the PR?

**Comment by RedBullDogE** [comment]
I think we need to move this function into separate class in the utils (see ex. httphandler.utils.UserDataValidator class)

**Review by IINamelessII — APPROVED** [review]

**Pull request merged**